### PR TITLE
Add support for logging to the functions SDK

### PR DESCRIPTION
**Pull request by conradgrobler**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_functions/sdk/oak_functions/src/lib.rs [hunk 1]
```diff
@@ -76,6 +76,19 @@ pub fn storage_get_item(key: &[u8]) -> Result<Option<Vec<u8>>, OakStatus> {
...
 }
 
+/// Writes a debug log message.
+///
+/// These log messages are considered sensitive, so will only be logged by the runtime if the
+/// `oak_unsafe` feature is enabled.
+///
+/// See [`write_log_message`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#write_log_message).
+pub fn log(message: &str) -> Result<(), OakStatus> {
+    let message = &format!("[Wasm] {}", message);
```
Comment (tiziano88) [inline oak_functions/sdk/oak_functions/src/lib.rs:86]: I don't think this should be done by the SDK, it belongs to the logging logic in the runtime.

File: oak_functions/sdk/oak_functions/src/lib.rs [hunk 2]
```diff
@@ -76,6 +76,19 @@ pub fn storage_get_item(key: &[u8]) -> Result<Option<Vec<u8>>, OakStatus> {
     }
 }
 
+/// Writes a debug log message.
+///
+/// These log messages are considered sensitive, so will only be logged by the runtime if the
+/// `oak_unsafe` feature is enabled.
+///
+/// See [`write_log_message`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#write_log_message).
+pub fn log(message: &str) -> Result<(), OakStatus> {
```
Comment (tiziano88) [inline oak_functions/sdk/oak_functions/src/lib.rs:85]: FYI this may make it nicer to use: https://doc.rust-lang.org/std/convert/trait.AsRef.html#examples

**Review by tiziano88 — APPROVED** [review]

File: oak_functions/sdk/oak_functions/src/lib.rs (hunk 2 above)
Comment (rbehjati) [inline oak_functions/sdk/oak_functions/src/lib.rs:85]: Could this be a macro taking any number of parameters and formatting them?

**Review by rbehjati — APPROVED** [review]
LGTM.

File: oak_functions/sdk/oak_functions/src/lib.rs (hunk 2 above)
Comment (conradgrobler) [reply oak_functions/sdk/oak_functions/src/lib.rs]: Thanks. Done.

File: oak_functions/sdk/oak_functions/src/lib.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_functions/sdk/oak_functions/src/lib.rs]: Moved.

File: oak_functions/sdk/oak_functions/src/lib.rs (hunk 2 above)
Comment (conradgrobler) [reply oak_functions/sdk/oak_functions/src/lib.rs]: Good idea. Done.

**Pull request merged**